src/python/test/test_weighted_rips.py

```diff
@@ -10,18 +10,17 @@
 
 from gudhi.weighted_rips_complex import WeightedRipsComplex
 from gudhi.point_cloud.dtm import DTM
-import numpy
+import numpy as np
 from scipy.spatial.distance import cdist
 import pytest
 
 def test_dtm_rips_complex():
-    pts = numpy.array([[2.0, 2], [0, 1], [3, 4]])
+    pts = np.array([[2.0, 2], [0, 1], [3, 4]])
     dist = cdist(pts,pts)
     dtm = DTM(2, q=2, metric="precomputed")    
     r = dtm.fit_transform(dist)
-    w_rips = WeightedRipsComplex(distance_mattix=dist, filtration_values=r)
+    w_rips = WeightedRipsComplex(distance_mattix=dist, weights=r)
     st = w_rips.create_simplex_tree(max_dimension=2)
-    diag = st.persistence()
-    assert diag == [(0, (1.5811388300841898, float("inf"))), (0, (1.5811388300841898, 2.699172818834085)), (0, (1.5811388300841898, 2.699172818834085))]
+    persistence_intervals0 = st.persistence_intervals_in_dimension(0)
```

Comment:
You still need to call `persistence()` so that the persistence is actually computed, before you can read it with persistence_intervals_in_dimension.